crates/common/consensus/src/execution_engine/mod.rs

```
@@ -0,0 +1,144 @@
+pub mod new_payload_request;
+
+use alloy_primitives::{Address, Bytes, B256, U256, U64};
+use alloy_rlp::{bytes, Buf, Decodable, Encodable, RlpDecodable, RlpEncodable, EMPTY_STRING_CODE};
+use anyhow::anyhow;
+use new_payload_request::NewPayloadRequest;
+use serde::{Deserialize, Serialize};
+use ssz_derive::{Decode, Encode};
+use tree_hash_derive::TreeHash;
+
+use crate::deneb::execution_payload::ExecutionPayload;
+
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize, Encode, Decode, TreeHash)]
+pub struct ExecutionEngine {}
+
+impl ExecutionEngine {
+    /// Return ``True`` if and only if ``execution_payload.block_hash`` is computed correctly.
+    pub fn is_valid_block_hash(
+        &self,
+        execution_payload: ExecutionPayload,
+        parent_beacon_block_root: B256,
+    ) -> bool {
+        execution_payload.block_hash == execution_payload.header_hash(parent_beacon_block_root)
+    }
+
+    /// Return ``True`` if and only if the version hashes computed by the blob transactions of
+    /// ``new_payload_request.execution_payload`` matches ``new_payload_request.versioned_hashes``.
+    pub fn is_valid_versioned_hashes(
+        self: ExecutionEngine,
+        new_payload_request: NewPayloadRequest,
+    ) -> anyhow::Result<bool> {
```

Would be nice if blob tx decoding fails, then return false.

---

Shouldn't the higher level code handle the error, doing this suggestion would hide errors no?